jobs/build.Jenkinsfile

```diff
@@ -532,6 +533,10 @@ def run_multiarch_jobs(arches, src_commit, version, cosa_img) {
                 string(name: 'PIPECFG_HOTFIX_REPO', value: params.PIPECFG_HOTFIX_REPO),
                 string(name: 'PIPECFG_HOTFIX_REF', value: params.PIPECFG_HOTFIX_REF)
             ]
+            // This really should never take more than 5 minutes. Having a
+            // timeout ensures we don't wait for a long time if we somehow
+            // missed the transition.
```

Comment:
This should probably have just a little bit more of the context from the commit message in it.

Reply:
Added some. Is that better?

Reply:
:) - not really. You added more comments to the `wait_until_locked_or_continue` function rather than adding the context here.

What I'm looking for right here is more along the lines of "we need to wait here until the lock is taken because if we don't then we introduce a race between the mArch job and the release job and we don't want the release job to win that race."

Another thing.. Maybe we just move this function call up into the only place it's needed, which is in the `params.EARLY_ARCH_JOBS` section. It's not really needed in the non `params.EARLY_ARCH_JOBS` case, but I'm OK with leaving it here to keep the code more generic.

Reply:
> Another thing.. Maybe we just move this function call up into the only place it's needed, which is in the `params.EARLY_ARCH_JOBS` section. It's not really needed in the non `params.EARLY_ARCH_JOBS` case, but I'm OK with leaving it here to keep the code more generic.

I guess the problem with that is we have to loop over the arches again. Wouldn't be terrible, but not great either.

Reply:
Yeah, I'd rather just leave it there. I like that it solves the theoretical case too. (And I think it's not that theoretical if you're only building the QEMU image. Compressing just that image takes a few minutes and uploading it just a few seconds; if the `git clone` is being slow for whatever reason in the `build-arch` job, I can imagine it getting overtaken by the release job.)

Reply:
new comment LGTM